src/server/src/main/java/io/cassandrareaper/storage/cassandra/SchemaMigrationLock.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2019-2019 The Last Pickle Ltd
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cassandrareaper.storage.cassandra;
+
+import io.cassandrareaper.ReaperApplicationConfiguration;
+import io.cassandrareaper.ReaperException;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.VersionNumber;
+import com.datastax.driver.core.utils.UUIDs;
+import jersey.repackaged.com.google.common.base.Preconditions;
+import org.cognitor.cassandra.migration.Database;
+import org.cognitor.cassandra.migration.MigrationRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class SchemaMigrationLock {
+
+  private static final UUID SCHEMA_MIGRATION_LOCK_ID = UUIDs.startOf(1);
+  private static final Logger LOG = LoggerFactory.getLogger(SchemaMigrationLock.class);
+  private static PreparedStatement lockSchemaMigrationPrepStmt;
+  private static PreparedStatement unlockSchemaMigrationPrepStmt;
+  private static PreparedStatement renewLockSchemaMigrationPrepStmt;
+  private final VersionNumber version;
+  private final Session session;
+  private final ReaperApplicationConfiguration config;
+
+  public SchemaMigrationLock(VersionNumber version, Session session, ReaperApplicationConfiguration config) {
+    this.version = version;
+    this.session = session;
+    this.config = config;
+    if (0 >= VersionNumber.parse("2.2").compareTo(version)) {
+      // no system_distributed keyspace before 2.2
+      lockSchemaMigrationPrepStmt = session.prepare("insert into system_distributed.parent_repair_history"
+          + " (parent_id, started_at)"
+          + " values(?, dateOf(now())) IF NOT EXISTS USING TTL ?");
+      unlockSchemaMigrationPrepStmt = session.prepare("DELETE FROM system_distributed.parent_repair_history"
+          + " WHERE parent_id = ?");
+      renewLockSchemaMigrationPrepStmt = session.prepare("update system_distributed.parent_repair_history USING TTL ?"
+          + " set started_at = dateOf(now()) where parent_id = ?");
+    }
+  }
+
+  public synchronized boolean isSchemaMigrationRequired(com.datastax.driver.core.Cluster cassandra)
```

Comment:
it's not clear why this method is inside `SchemaMigrationLock`, it looks like logic around schema migration that's only trying to optimise when to grab the lock. but that complicates/breaks things.

can't the sleep retry lock loop be moved inside the `lockSchemaMigration()`  method.
that would mean the lock is taken and released every reaper startup, but i think we can live with that as this is intended as a temporary hack in the code.

Reply:
this is what i mean: https://github.com/thelastpickle/cassandra-reaper/commit/eb7a757ba38be468f84c191197be2f9ba915ebfe